test/t/io/test_writer.cpp

```diff
@@ -111,3 +111,43 @@ TEST_CASE("Writer") {
 
 }
 
+TEST_CASE("Writer with user-provided Pool") {
+    osmium::io::Header header;
+    header.set("generator", "test_writer.cpp");
+
+    osmium::io::Reader reader{with_data_dir("t/io/data.osm")};
+    osmium::memory::Buffer buffer = reader.read();
+    REQUIRE(buffer);
+    REQUIRE(buffer.committed() > 0);
+
+    SECTION("Successful writes") {
```

Comment:
These tests are just about the `Pool` so that's okay if there is nothing failing in here, we don't need to say this explicitly.